session/amount_calculator_test.go

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package session
+
+import (
+	"testing"
+	"time"
+
+	"github.com/mysteriumnetwork/node/money"
+	"github.com/mysteriumnetwork/node/services/openvpn/discovery/dto"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_CorrectMoneyValueIsReturnedForTotalAmount(t *testing.T) {
+	aCalc := AmountCalc{
+		PaymentDef: dto.PaymentPerTime{
+			Duration: time.Minute,
+			Price: money.Money{
+				Amount:   100,
+				Currency: money.CURRENCY_MYST,
+			},
+		},
+	}
+
+	elapsed := time.Minute*3 + time.Second*15
+
+	totalAmount := aCalc.TotalAmount(elapsed)
+
+	assert.Equal(t, uint64(300), totalAmount.Amount)
```

Comment:
Do I understand correctly, that we have 100 MYST / min price here, and duration of 3 min and 15 sec? Why we expect 300 price and not 325 ?

Reply:
Because we are calcalting price per minute, not per second and these seconds a dropped in favour or consumer.

Reply:
Can this be abused by the user by stopping session every 59 seconds, and starting a new one?

Reply:
In theory - yes. In reality, you'll be asked for a promise rather quickly by the provider and he'll be killing the connection if you don't provide one. You could get a few seconds for free, probably.

Reply:
Not actually - as consumer is required to pay in advance to keep balance positive

Reply:
So the actual elapsed time cannot cross billing unit boundary? In this case 3 min.